crates/viewer/re_time_panel/src/time_panel.rs

```diff
@@ -1885,163 +1905,115 @@ fn copy_time_properties_context_menu(ui: &mut egui::Ui, time: TimeReal) {
-            x = time_ranges_ui
-                .x_from_time_f32(time)
-                .unwrap_or(pointer_pos.x);
-        }
+    let hovered_time = if response.hovered()
+        && let Some(pointer_pos) = pointer_pos
+    {
+        time_ranges_ui.snapped_time_from_x(ui, pointer_pos.x)
+    } else {
+        None
+    };
 
-        ui.paint_time_cursor(
-            time_area_painter,
-            &response,
-            x,
-            Rangef::new(timeline_rect.top(), ui.max_rect().bottom()),
-        );
+    // Press to move time:
+    if ui.input(|i| i.pointer.primary_down())
+        // `interact_pointer_pos` is set as soon as the mouse button is down on it,
+        // without having to wait for the drag to go far enough or long enough
+        && response.interact_pointer_pos().is_some()
+        && let Some(time) = hovered_time
+    {
+        time_commands.push(TimeControlCommand::SetTime(time));
+        time_commands.push(TimeControlCommand::Pause);
     }
 
-    // "click here to view time here"
-    if let Some(pointer_pos) = pointer_pos {
-        let is_pointer_in_time_area_rect =
-            ui.ui_contains_pointer() && time_area_painter.clip_rect().contains(pointer_pos);
-        let is_pointer_in_timeline_rect =
-            ui.ui_contains_pointer() && timeline_rect.contains(pointer_pos);
-
-        let on_timeline = !is_hovering_time_cursor
-            && !time_area_double_clicked
-            && is_pointer_in_time_area_rect
-            && !is_anything_being_dragged
-            && !is_hovering_the_loop_selection;
-
-        if on_timeline {
-            ui.ctx().set_cursor_icon(timeline_cursor_icon);
-        }
-
-        // Click to move time here:
-        let time_area_response = ui.interact(
-            time_area_painter.clip_rect(),
-            ui.id().with("time_area_painter_id"),
-            egui::Sense::click(),
-        );
-
-        let hovered_time = if time_area_response.hovered() {
-            time_ranges_ui.snapped_time_from_x(ui, pointer_pos.x)
-        } else {
-            None
-        };
-
-        if !is_hovering_the_loop_selection {
-            let mut set_time_to_pointer = || {
-                if let Some(time) = hovered_time {
-                    let time = time_ranges_ui.clamp_time(time);
-                    time_commands.push(TimeControlCommand::SetTime(time));
-                    time_commands.push(TimeControlCommand::Pause);
-                }
-            };
-
-            // click on timeline = set time + start drag
-            // click on time area = set time
-            // double click on time area = reset time
-            if !is_anything_being_dragged
-                && is_pointer_in_timeline_rect
-                && ui.input(|i| i.pointer.primary_down())
-            {
-                set_time_to_pointer();
-                ui.ctx().set_dragged_id(time_drag_id);
-            } else if is_pointer_in_time_area_rect {
-                if time_area_response.double_clicked() {
```

Comment:
IIUC think this is the only place we actually did the double click to reset, which is now broken!

Reply:
Ahhh, okay makes sense with these other changes.